model-optimizer/extensions/middle/SplitConcatPairToInterpolate.py

```diff
@@ -32,6 +32,19 @@
 
 
 def get_concat_after_split(split: Node) -> Optional[Node]:
+    """
+    This function gets consumers of the 'split' node, checks that the following conditions are fulfilled:
+    1) 'split' node has only one consumer;
+    2) for any output port of 'split', number of corresponding input ports of the consumer is the same;
+    3) for any output port 'i' of the 'split', corresponding input ports of the consumer are
+       [i * m, ..., i * m + (m - 1)], where 'm' is the same for all 'i';
+    4) the consumer operation is 'Concat';
+    5) 'split' is a unique producer for this 'Concat';
+    and, if all these conditions are fulfilled, returns the above mentioned 'Concat' node. Otherwise, if some of these
+    conditions is false, this functions returns None.
+    :param split: Split node
+    :return: Concat node, if all conditions are fulfilled, or None otherwise
```

Comment:
nit: `None != nothing` so I propose to change: Concat node, if all conditions are fulfilled, or nothing otherwise

Reply:
What does it mean `nothing`? According to type annotation in the header of the function, its returned value type is `typing.Optional[Node]`. The Python3 module `typing` defines `Optional[T]` as `Union[T, None]` (see https://docs.python.org/3/library/typing.html#typing.Optional). Hence, your comment is not correct.

Reply:
`return` as you do now is not equivalent to `return None`.

Reply:
What? According to https://docs.python.org/3/reference/simple_stmts.html#the-return-statement, 

> return may only occur syntactically nested in a function definition, not within a nested class definition.
If an expression list is present, it is evaluated, else None is substituted.
return leaves the current function call with the expression list (or None) as return value.

And this function is not generator function.

Hence, your comment is not correct.